outputs.tf

```diff
@@ -1,16 +1,16 @@
 output "instance" {
   description = "List of instances IDs created by this module"
-  value       = "${module.dcos-bootstrap-instance.instances[0]}"
+  value       = "${element(concat(module.dcos-bootstrap-instance.instances, list("")), 0)}"
 }
 
 output "public_ip" {
   description = "List of public ip addresses created by this module"
-  value       = "${module.dcos-bootstrap-instance.public_ips[0]}"
+  value       = "${element(concat(module.dcos-bootstrap-instance.public_ips, list("")), 0)}"
 }
 
 output "private_ip" {
   description = "List of private ip addresses created by this module"
-  value       = "${module.dcos-bootstrap-instance.private_ips[0]}"
+  value       = "${element(concat(module.dcos-bootstrap-instance.private_ips, list("")), 0)}"
```

Comment:
oh I see the concat is meant to be used while destroying... do I get this right?

Reply:
when the bootstrap is conditionally created in the case of the multi-region environment where it pull the bootstrap from a remote region, having an empty object in an array throws an error. More info on this here: https://github.com/hashicorp/hil/issues/50

The workaround in terraform is to always add a list("") and have it return that instead in the event an empty list exist.

Reply:
And this specific comment as well https://github.com/hashicorp/terraform/issues/16580#issuecomment-342573652